frontend/hub/administration/repositories/RepositoryForm.tsx

```diff
@@ -0,0 +1,309 @@
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router-dom';
+import {
+  LoadingPage,
+  PageFormCheckbox,
+  PageFormSelect,
+  PageFormTextInput,
+  PageHeader,
+  PageLayout,
+  useGetPageUrl,
+  usePageNavigate,
+} from '../../../../framework';
+import { HubPageForm } from '../../common/HubPageForm';
+import { HubRoute } from '../../main/HubRoutes';
+import { pulpAPI } from '../../common/api/formatPath';
+import { IRemotes } from './../remotes/Remotes';
+import { useParams } from 'react-router';
+import { PageFormGroup } from '../../../../framework/PageForm/Inputs/PageFormGroup';
+import { useGet } from '../../../common/crud/useGet';
+import { PulpItemsResponse } from '../../common/useHubView';
+import { Repository } from './Repository';
+import { Label } from '@patternfly/react-core';
+import { PageFormAsyncSelect } from '../../../../framework/PageForm/Inputs/PageFormAsyncSelect';
+import { useCallback, ReactNode } from 'react';
+import { useSelectRemoteSingle } from './hooks/useRemoteSelector';
+import { useRepositoryBasePath, parsePulpIDFromURL } from '../../common/api/hub-api-utils';
+import { postHubRequest, putHubRequest } from '../../common/api/request';
+import { postRequest } from '../../../common/crud/Data';
```

Comment:
Is there a particular reason to use `postRequest` over `postHubRequest`? Just to learn.

Reply:
No reason. Fixed :)